flexmeasures/data/models/planning/storage.py

```diff
@@ -693,54 +739,75 @@ def deserialize_flex_config(self):
+
+            self.ensure_soc_min_max()
+
+            # Now it's time to check if our flex configuration holds up to schemas
+            self.flex_model = StorageFlexModelSchema(
+                start=self.start,
+                sensor=self.sensor,
+                default_soc_unit=self.flex_model.get("soc-unit"),
+            ).load(self.flex_model)
+
+            # Extend schedule period in case a target exceeds its end
+            self.possibly_extend_end(soc_targets=self.flex_model.get("soc_targets"))
+        elif isinstance(self.flex_model, list):
+            self.flex_model = SequentialFlexModelSchema(many=True).load(self.flex_model)
+            for d, sensor_flex_model in enumerate(self.flex_model):
+                self.flex_model[d] = StorageFlexModelSchema(
+                    start=self.start, sensor=sensor_flex_model["sensor"]
+                ).load(sensor_flex_model["sensor_flex_model"]["sensor-flex-model"])
+                self.flex_model[d]["sensor"] = sensor_flex_model["sensor"]
+            # Extend schedule period in case a target exceeds its end
+            for child_flex_model in self.flex_model:
+                self.possibly_extend_end(
+                    soc_targets=child_flex_model.get("soc_targets"),
+                    sensor=child_flex_model["sensor"],
                 )
-            else:
-                self.flex_model["soc-at-start"] = 0
-
-        self.ensure_soc_min_max()
-
-        # Now it's time to check if our flex configuration holds up to schemas
-        self.flex_model = StorageFlexModelSchema(
-            start=self.start,
-            sensor=self.sensor,
-            default_soc_unit=self.flex_model.get("soc-unit"),
-        ).load(self.flex_model)
-        self.flex_context = FlexContextSchema().load(self.flex_context)
 
-        # Extend schedule period in case a target exceeds its end
-        self.possibly_extend_end()
+        else:
+            raise TypeError(
+                f"Unsupported type of flex-model: '{type(self.flex_model)}'"
+            )
 
         return self.flex_model
 
-    def possibly_extend_end(self):
+    def possibly_extend_end(self, soc_targets, sensor: Sensor = None):
         """Extend schedule period in case a target exceeds its end.
 
         The schedule's duration is possibly limited by the server config setting 'FLEXMEASURES_MAX_PLANNING_HORIZON'.
 
         todo: when deserialize_flex_config becomes a single schema for the whole scheduler,
               this function would become a class method with a @post_load decorator.
         """
-        soc_targets = self.flex_model.get("soc_targets")
+        if sensor is None:
+            sensor = self.sensor
 
         if soc_targets and not isinstance(soc_targets, Sensor):
             max_target_datetime = max([soc_target["end"] for soc_target in soc_targets])
             if max_target_datetime > self.end:
-                max_server_horizon = get_max_planning_horizon(self.resolution)
```

Comment:
My first thought is that this function still does what it is supposed to do, namely extending the planning horizon if needed, therein taking into account the resolution of each device's power sensor. But it deserves a closer look, especially regarding how the horizon is derived using the resolution.

Reply:
Yeah, looks good.